### PR TITLE
IDEM-159

### PR DESCRIPTION
**Pull request by JasminFarhi**
Image from iOS (121)
Image from iOS (120)

### CODE REVIEW

**Review by amagon270 — APPROVED** [review]

File: app/src/context/Exchange.tsx [hunk 1]
```diff
@@ -162,7 +162,11 @@ export const ExchangeProvider: React.FC<{
       }
     } catch (error: any) {
       console.log(error);
-      Alert.alert(error.response.data);
+      if (error.response.data === "Username or password is incorrect") {
+        Alert.alert("Invalid password");
+      } else {
+        Alert.alert(error.response.data);
+      }
```
Comment (bitcoinbrisbane) [inline app/src/context/Exchange.tsx:169]: Reduce the complexity by removing the else.

```suggestion
      }
      Alert.alert(error.response.data);
```

File: app/src/context/Exchange.tsx [hunk 2]
```diff
@@ -162,7 +162,11 @@ export const ExchangeProvider: React.FC<{
       }
     } catch (error: any) {
       console.log(error);
-      Alert.alert(error.response.data);
+      if (error.response.data === "Username or password is incorrect") {
```
Comment (bitcoinbrisbane) [inline app/src/context/Exchange.tsx:165]: I thought i said the stand up that this error deoesnt make sense as the username / email HAS to be correct.
```suggestion
      if (error.response.data === "The password you have provided for this account is incorrect") {
```

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Few things to look at.  Also, the title "Share Details" doesn't make sense.

File: app/src/context/Exchange.tsx (hunk 1 above)
Comment (JasminFarhi) [reply app/src/context/Exchange.tsx:169]: If I remove the else it will show two alerts...

Comment (bitcoinbrisbane) [reply app/src/context/Exchange.tsx:169]: Fair.

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Pull request merged**